lib/src/components/NxFileUpload/NxMultiFileUpload/__tests__/NxMultiFileUpload.test.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React from 'react';
+import { render } from '@testing-library/react';
+import { pipe } from 'ramda';
+
+import { rtlRender, rtlRenderElement } from '../../../../__testutils__/rtlUtils';
+import NxMultiFileUpload from '../NxMultiFileUpload';
+import NxForm from '../../../NxForm/NxForm';
+
+describe('NxMultiFileUpload', function() {
+  const minimalProps = {
+        files: null,
+        onChange: () => {}
+      },
+      quickRender = rtlRender(NxMultiFileUpload, minimalProps),
+      renderEl = rtlRenderElement(NxMultiFileUpload, minimalProps),
+      renderInput = pipe(renderEl, el => el?.querySelector('input[type=file]'));
+
+  it('renders a file input and a button', function() {
+    const component = quickRender(),
+
+        // file inputs have no aria role and the label text is outside of the scope of this component,
+        // so we have to query like this
+        input = component.container.querySelector('input[type=file]'),
+        button = component.queryByRole('button', { name: 'Choose File' });
+
+    expect(input).toBeDefined();
+    expect(button).toBeDefined();
+
+    // cannot effectively test that the button activates the input here, will do in functional test
+  });
+
+  it('sets specified ClassNames on the top-level element', function() {
+    const component = renderEl()!;
+    const customComponent = renderEl({ className: 'foo'});
+
+    expect(customComponent).toHaveClass('foo');
+
+    for (const cls of Array.from(component.classList)) {
+      expect(customComponent).toHaveClass(cls);
+    }
+  });
+
+  it('sets specified attributes on the input', function() {
+    const component = renderEl({id: 'bar', lang: 'en' })!,
+        input = component.querySelector('input[type=file]');
+
+    expect(input).toHaveAttribute('id', 'bar');
+    expect(input).toHaveAttribute('lang', 'en');
+  });
+
+  it('sets the multiple attribute on the input', function() {
+    const component = renderEl()!,
+        input = component.querySelector('input[type=file]');
+
+    expect(input).toHaveAttribute('multiple');
+  });
+
+  it('renders a "No file selected" message if files is null', function() {
+    expect(renderEl()).toHaveTextContent('No file selected');
+  });
+
+  it('render a "No File Selected" aria-label if files is null', function() {
```

Comment:
As mentioned in a comment on the impl, I think this behavior is wrong

Reply:
> As mentioned in a comment on the impl, I think this behavior is wrong

Fixed here with testing `toHaveAccessibleDescription` instead - [aa1820b](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/aa1820b58cb14aa746f4ef038b893f96b786fe30)